gym/envs/box2d/car_racing.py

```diff
@@ -432,11 +441,16 @@ def reset(
         else:
             return self.step(None)[0], {}
 
-    def step(self, action: np.ndarray):
+    def step(self, action: Union[np.ndarray, int]):
         if action is not None:
-            self.car.steer(-action[0])
-            self.car.gas(action[1])
-            self.car.brake(action[2])
+            if self.continuous:
+                self.car.steer(-action[0])
+                self.car.gas(action[1])
+                self.car.brake(action[2])
+            else:
+                self.car.steer(-1 * (action == 1) + 1 * (action == 2))
+                self.car.gas(0.2 * (action == 3))
+                self.car.brake(0.8 * (action == 4))
```

Comment:
I actually started with values from [here](https://notanymike.github.io/Solving-CarRacing/), but braking and accelerating at the same time as an action doesn't seem intuitive, so I changed it to a value that showed good learning performance.

As for steering, I started with values of 1.0 (which was harder to learn since every turn at speed caused the car to spinout), so I reduced them from 1.0 to 0.8 to 0.6, which heuristically showed good learning performance.

Admittedly there's no real science to it.

Reply:
Is this something that we could have as hyperparameters that users could change?

Reply:
I think that is acceptable. You could have the arguments are keyword only [this](https://lukeplant.me.uk/blog/posts/keyword-only-arguments-in-python/)
So only people who intended to use it, can. 
Or you could the magic numbers are attributes of the environment that are assessable to the user + documentation to explain how to modify them?

Reply:
Keyworded arguments are pretty cool, I think that's what I'll use.
In general having env parameters that are accessible to outside users are a pretty bad idea as these should really stay as private attributes (in fact most env functions are treated as such and are not callable from the outside).

Personally I'm inclined to just add documentation on how to make a custom wrapper that show the conversion of continuous to discrete actions. If everyone agrees that this is the right way to go, then I'll close this PR and re-open another one with the documentation change instead. :)

Reply:
Looking back at my message, I agree the custom wrapper is better

Reply:
I talked to @jkterry1 about the possibility of using a wrapper a few days ago and he believes that CarRacing should support discrete actions natively (if I understood him correctly).

What about registering a factory function that produces wrapped CarRacing environments? That function could then accept the configuration for the conversion to continuous actions. That way people can use `make` to generate discrete CarRacing environments, but there are no additional kwargs for the standard CarRacing environment.

Reply:
@Markus28 My understanding was to do both though reading that back it is not clear. 
My thinking was to have a discrete version (without a wrapper) that has magic numbers that are known to allow the car racing to train successfully. As @jjshoots has implemented currently.
Then in the documentation a section explaining how a custom wrapper that translates discrete actions to the continuous action space could be done. To allow people to use different numbers

Reply:
Hi, been MIA for awhile, if I understand the conversation correctly, what should happen here is to add documentation explaining how to convert the continuous env into a discrete one using wrappers, as well as keeping the current discrete action implementation in place?